core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorTableMetadata.java

```diff
@@ -37,19 +39,26 @@ public ConnectorTableMetadata(SchemaTableName table, List<ColumnMetadata> column
 
     public ConnectorTableMetadata(SchemaTableName table, List<ColumnMetadata> columns, Map<String, Object> properties)
     {
-        this(table, columns, properties, Optional.empty());
+        this(table, columns, properties, Optional.empty(), emptyList());
     }
 
     public ConnectorTableMetadata(SchemaTableName table, List<ColumnMetadata> columns, Map<String, Object> properties, Optional<String> comment)
+    {
+        this(table, columns, properties, comment, emptyList());
+    }
+
+    public ConnectorTableMetadata(SchemaTableName table, List<ColumnMetadata> columns, Map<String, Object> properties, Optional<String> comment, List<String> checkConstraints)
```

Comment:
per @martint  https://github.com/trinodb/trino/pull/14863#discussion_r1013150331
let's call it "integrity constraints".

Reply:
Integrity constraints contain unique & referential constraints in addition to check. I would introduce a new class if necessary. 
```
<table constraint> ::=
    <unique constraint definition> 
  | <referential constraint definition> 
  | <check constraint definition> 
```

Reply:
👍  let's stay with "check"